diff.py

```diff
@@ -1318,12 +1330,14 @@ def dump_binary(
         end_addr = eval_int(end, "End address must be an integer expression.")
     else:
         end_addr = start_addr + config.max_function_size_bytes
-    objdump_flags = ["-Dz", "-bbinary"] + ["-EB" if config.arch.big_endian else "-EL"]
+    objdump_flags = ["-Dz"] + ["-belf32-littlearm" if config.arch.name == "armel" else "-bbinary"] + ["-EB" if config.arch.big_endian else "-EL"]
```

Comment:
This is somewhat of an abuse of `dump_binary`, making it not dump a binary range at all but rather parse an ELF file... I'm confused as to how this works when you're also passing in --start-address= and --stop-address=. Can you explain your workflow? Any chance you can get your project set up to use dump_objfile or dump_elf instead, which should be more user-friendly? e.g. dump_objfile/-o accepts a function name as argument which is nicer than a range of bytes.

Reply:
For our project, we need to control the global ordering of functions at the final link stage, by essentially hard-coding the ordering. The best way for us to do this is to specify `--split_sections` to the compiler so that it generates an elf section for each function, then use a linkerscript to order those sections into the global `.text` section. So, each function `foo` gets placed in an elf section called `i.foo`. Then, "non-matchings" use the same section names, so that the linker script doesnt need to be edited to replace a non-matching with decompiled code.

Anyway, I was able to tweak our project setup to make it work with dump_elf, or maybe dump_obj with a bit more work, which is better.